ui/src/parts/lightbox.tsx

```diff
@@ -0,0 +1,224 @@
+// Copyright 2020 H2O.ai, Inc.
+//
+// Licensed under the Apache License, Version 2.0 (the "License");
+// you may not use this file except in compliance with the License.
+// You may obtain a copy of the License at
+//
+//     http://www.apache.org/licenses/LICENSE-2.0
+//
+// Unless required by applicable law or agreed to in writing, software
+// distributed under the License is distributed on an "AS IS" BASIS,
+// WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+// See the License for the specific language governing permissions and
+// limitations under the License.
+
+import { box, Box, S, U } from 'h2o-wave'
+import React from 'react'
+import { stylesheet, style } from 'typestyle'
+import * as Fluent from '@fluentui/react'
+import { clas, cssVar } from '../theme'
+import { getColorFromString, isDark } from '@fluentui/react'
+import { getImageSrc } from '../image'
+
+export const lightboxB: Box<LightboxProps | null> = box()
+
+const
+  HEADER_HEIGHT = 40,
+  IMAGE_CAPTIONS_HEIGHT = 60,
+  IMAGE_NAV_HEIGHT = 142
+
+const
+  css = stylesheet({
+    body: {
+      position: 'fixed',
+      inset: '0px',
+      zIndex: 1,
+      backgroundColor: 'rgba(0, 0, 0, 0.9)',
+    },
+    img: {
+      maxWidth: '100%',
+      objectFit: 'scale-down',
+    },
+    header: {
+      height: HEADER_HEIGHT,
+      textAlign: 'right'
+    },
+    content: {
+      display: 'flex',
+      position: 'relative',
+      justifyContent: 'center',
+    },
+    footer: {
+      textAlign: 'center',
+      color: '#fff',
+    },
+    imageNav: {
+      height: IMAGE_NAV_HEIGHT,
+      overflow: 'auto',
+      whiteSpace: 'nowrap'
+    },
+    navImg: {
+      boxSizing: 'border-box',
+      objectFit: 'cover',
+      height: '120px',
+      width: '120px',
+      margin: '0px 2px',
+      cursor: 'pointer',
+      filter: 'brightness(30%)',
+      border: '2px solid #000'
+    },
+    imgCaptions: {
+      whiteSpace: 'nowrap',
+      padding: '10px 40px',
+      height: IMAGE_CAPTIONS_HEIGHT - 20
```

Comment:
What's this magic number `20`?

Reply:
Isn't it self-explanatory when there is a top/bottom 10px padding?

Reply:
You see it's not since I am asking about it :D Have a look at `box-sizing` CSS prop, that's the proper way to do this instead of calculations.

Reply:
I was not sure if you don't know or you just want to make code even better. Thanks for the `box-sizing` prop, that is what I was looking for.